### ISSUE TITLE
Allow for always-below

### ISSUE THREAD

**Issue by lf-** [LemonBoy/bar#204]
Currently lemonbar always makes itself always on top, however that messes up with fullscreen windows since the bar is always above them. _NET_WM_STATE_BELOW looks like what is necessary to force the bar window to the bottom.

And this is not a WM issue. bar is an unmanaged window.

**Comment by okraits** [comment]
I don't have that issue here with openbox.

**Comment by lf-** [comment]
```
    'lemonbar',
    '-fSource Sans Pro:size=11',
    '-fFontAwesome:size=11',
    '-gx20'
```

I don't believe that I'm using the -d switch.

**Comment by okraits** [comment]
I didn't say that you use the -d switch, I just mentioned what I experienced.

**Comment by ThomasAdam** [comment]
With unmanaged windows, it's irrelevant.

**Comment by LemonBoy** [comment]
Uhh, I think it's up to the WM to raise the fullscreen window to the top of the stack.

**Comment by lf-** [comment]
I'd like to eliminate this script:

```bash
#!/bin/bash

until bar_id=$(xdo id -a aiopanel) ; do
    sleep 0.01
done

xdo below -t $(xdo id -n root) $bar_id
```

If the bar could do this itself, that would be awesome.

**Comment by LemonBoy** [comment]
Holy shit, that's horrid.
Did you try changing [this](https://github.com/LemonBoy/bar/blob/master/lemonbar.c#L733) line instead of coming up with that monstrosity?

**Comment by lf-** [comment]
Yes, I tried that before making a hack script. It didn't seem to change behavior.

**Comment by LemonBoy** [comment]
What about this patch:

```diff
diff --git a/lemonbar.c b/lemonbar.c
index 9220af0..d468dad 100644
--- a/lemonbar.c
+++ b/lemonbar.c
@@ -1210,6 +1210,8 @@ init (char *wm_name)
         // Some WM such as Openbox need this
         xcb_configure_window(c, mon->window, XCB_CONFIG_WINDOW_X | XCB_CONFIG_WINDOW_Y, (const uint32_t []){ mon->x, mon->y });
 
+        xcb_configure_window(c, mon->window, XCB_CONFIG_WINDOW_STACK_MODE, (const uint32_t []){ XCB_STACK_MODE_BELOW });
+
         // Set the WM_NAME atom to the user specified value
         if (wm_name)
             xcb_change_property(c, XCB_PROP_MODE_REPLACE, mon->window, XCB_ATOM_WM_NAME, XCB_ATOM_STRING, 8 ,strlen(wm_name), wm_name);
```

**Comment by ajstensland** [comment]
Applying the above patch worked perfectly for me. (i3-gaps, with the -d option)

**Comment by LemonBoy** [comment]
Fixed by #250
